v2/internal/controllers/crd_dataprotection_backuppolicy_test.go

```diff
@@ -0,0 +1,218 @@
+								ObjectType: &AzureBackupRule_ObjectType,
+								BackupParameters: &dataprotection.BackupParameters{
+									AzureBackupParams: &dataprotection.AzureBackupParams{
+										BackupType: &AzureBackupParams_BackupType_Value,
+										ObjectType: &AzureBackupParams_ObjectType_Value,
+									},
+								},
+								DataStore: &dataprotection.DataStoreInfoBase{
+									DataStoreType: &DataStore_DataStoreType_Value,
+									ObjectType:    &DataStore_ObjectType_Value,
+								},
+								Trigger: &dataprotection.TriggerContext{
+									Schedule: &dataprotection.ScheduleBasedTriggerContext{
+										ObjectType: &Schedule_ObjectType_Value,
+										Schedule: &dataprotection.BackupSchedule{
+											RepeatingTimeIntervals: []string{"R/2023-06-07T10:26:32+00:00/PT4H"},
+											TimeZone:               &Schedule_Timezone_Value,
+										},
+										TaggingCriteria: []dataprotection.TaggingCriteria{
+											{
+												IsDefault:       &TaggingCriteria_isDefault_Value,
+												TaggingPriority: &TaggingCriteria_TaggingPriority_Value,
+												TagInfo: &dataprotection.RetentionTag{
+													TagName: &TaggingCriteria_TagInfo_TagName_Value,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						{
+							AzureRetention: &dataprotection.AzureRetentionRule{
+								Name:       &AzureRetentionRule_Name,
+								ObjectType: &AzureRetentionRule_ObjectType,
+								IsDefault:  &AzureRetentionRule_IsDefault,
+								Lifecycles: []dataprotection.SourceLifeCycle{
+									{
+										DeleteAfter: &dataprotection.DeleteOption{
+											AbsoluteDeleteOption: &dataprotection.AbsoluteDeleteOption{
+												Duration:   &AzureRetentionRule_Lifecycles_DeleteAfter_Duration,
+												ObjectType: &AzureRetentionRule_Lifecycles_DeleteAfter_ObjectType,
+											},
+										},
+										SourceDataStore: &dataprotection.DataStoreInfoBase{
+											DataStoreType: &AzureRetentionRule_Lifecycles_SourceDataStore_DataStoreType,
+											ObjectType:    &AzureRetentionRule_Lifecycles_SourceDataStore_ObjectType,
+										},
+										TargetDataStoreCopySettings: []dataprotection.TargetCopySetting{},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tc.CreateResourceAndWait(backuppolicy)
+
+	// Assertions and Expectations
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.DatasourceTypes).To(BeEquivalentTo([]string{"Microsoft.ContainerService/managedClusters"}))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.ObjectType).To(BeEquivalentTo(&backupPolicy_ObjectType))
+
+	// Assertions and Expectations for BackupPolicy:AzureBackupRule
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.Name).To(BeEquivalentTo(&AzureBackRule_Name))
+	tc.Expect(backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0].AzureBackup.ObjectType).To(BeEquivalentTo(&AzureBackupRule_ObjectType))
```

Comment:
Suggest pulling the rule out into a local variable to make these easier to read:

```suggestion
        policyRule0 := backuppolicy.Status.Properties.BackupPolicy.PolicyRules[0]
	tc.Expect(policyRule0.AzureBackup.Name).To(BeEquivalentTo(&AzureBackRule_Name))
	tc.Expect(policyRule0.AzureBackup.ObjectType).To(BeEquivalentTo(&AzureBackupRule_ObjectType))
```

This also makes it clearer that you have multiple assertions on the one object. 

Similar extraction for the other assertions below would be good too.

Reply:
Made the suggested changes!